### PR TITLE
Update Babel Polyfills

### LINKED ISSUE #641: Update Babel Polyfills [13pt]
Update Babel `babel-preset-env` to the last version to fix the problems we had with polyfills in old browsers.

### LINKED ISSUE #782: No colors appearing in terminal
In webpack-dev-middleware 3.7.2 it worked fine, but in version 4.0.2 it's showing <i> tag, <e> tag.

- Operating System: Ubuntu 20.04
- Node Version: 14.15.1
- NPM Version: 6.14.8
- webpack Version: 5.6.0
- webpack-dev-middleware Version: 4.0.2

### Expected Behavior
Show correct colors in terminal
image

### Actual Behavior
image

### PR DESCRIPTION
**Pull request by luisherranz**
**What**:

I have updated our Babel configuration to make it work with the latest versions of Babel.

**Why**:

We were not injecting the polyfills correctly and some of them were missing, both in the modules and es5 versions.

**How**:

- I have updated `babel-preset-env` to the latest version, which relies on `core-js` v3 instead of `@babel/polyfill`.
- I extracted a list of all the polyfills added to both `es5` and `module`.
- I switched `useBuiltIns` from `"entry"` to `"usage"` in `babel-preset-env` so only polyfills that are actually used in the code are added. That means we don't need to add any entry to our files.
- I had to exclude `webpack`, `core-js` and `regenerator-runtime` from babel itself because if they are transpiled they don't work.
- I had to switch to `sourceType: unambiguous` to make this new configuration work.
- I have updated the target version of node to 10 (previously it was 8).
- Now both `module` and `node` targets use core-js. Previously they didn't.
- I have added the `bugfixes` option because it will be the default in Babel 8.
- I have updated the packages of `@frontity/core`.
- I have updated the packages of `frontity`.

**Tasks**:

- [x] Code
- [x] TSDocs
- [x] Add a changeset (with link to its [Feature Discussion](https://community.frontity.org/c/33) if it exists)

**Unrelated Tasks**

- [ ] Update `webpack-dev-middleware` once this bug is fixed: https://github.com/webpack/webpack-dev-middleware/issues/782
- [ ] TypeScript
- [ ] Unit tests
- [ ] End to end tests
- [ ] TypeScript tests
- [ ] Update starter themes
- [ ] Update other packages
- [ ] Open an issue for this feature in the [docs repo](https://github.com/frontity/docs/wiki/Code-Releases)
- [ ] Update community discussions

**Additional Comments**

There is a bug right now in the latest version of `webpack-dev-middleware` that outputs `<i>` in the console: https://github.com/webpack/webpack-dev-middleware/issues/782

Screenshot 2020-12-08 at 13 45 17

As soon as it is fixed, I will update our version.

### CODE REVIEW

**Comment by luisherranz** [comment]
> Update webpack-dev-middleware once this bug is fixed: webpack/webpack-dev-middleware#782

Let's merge this as it is for now and I will add another PR once this is solved.

**Comment by luisherranz** [comment]
@cristianbote I have assigned this review to you so we can merge it before finishing your React 17 PR (https://github.com/frontity/frontity/pull/661).

I made a quick video to explain the work here: https://www.loom.com/share/6a487951c5c04f8ca61cdb4d70fbc820

If you have questions let me know! 🙂

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "prettier": "^2.0.1",
     "replace-in-files-cli": "^1.0.0",
     "ts-jest": "^26.2.0",
-    "typescript": "^4.0.5"
+    "typescript": "4.0.5"
```
Comment (cristianbote) [inline package.json:38]: Small wondering, not blocking or anything, but is the version lock needed?

**Review by cristianbote — APPROVED** [review]
Now I get why you've said that we've been working on the same changes 😄 

It's all good from my side!

File: package.json (hunk 1 above)
Comment (luisherranz) [reply package.json:38]: It seems like a problem with `ts-jest` not supporting TS 4.1. [I recorded a video to explain it](https://www.loom.com/share/e708cd1c072b4d9e8034a074e331aaf6) and I was planning to do a separate PR if you guys have seen the same in your branches.

**Comment by luisherranz** [comment]
Thanks Cristian!

**Comment by luisherranz** [comment]
I have opened a bug for the `<i>` problem: https://github.com/frontity/frontity/issues/662

**Pull request merged**